core/src/main/scala/org/http4s/ResponseCookie.scala

```diff
@@ -63,7 +63,7 @@ class RequestCookieJar(private val headers: Seq[Cookie])
   def get(key: String): Option[Cookie] = headers.find(_.name == key)
   def apply(key: String): Cookie = get(key).getOrElse(default(key))
   def contains(key: String): Boolean = headers.exists(_.name == key)
-  def getOrElse(key: String, default: => String): Cookie = get(key).getOrElse(Cookie(key, default))
+  def getOrElse(key: String, default: => String): Cookie = get(key).getOrElse(RequestCookie(key, default))
```

Comment:
Doesn't this whole Cookie jar pertain to `RequestCookie`s?

This cookie jar is full of unsafe methods, isn't particularly useful to a client maintaining an active conversation, etc.  I think it's a candidate for deprecation and/or serious improvement.  In any case, I would also move it to a `RequestCookieJar` file.